templates/commands/goldentest/test_funcs.go

```diff
@@ -254,16 +264,20 @@ func renameGitDirsAndFiles(dir string) error {
 	return nil
 }
 
-func parseTemplateLocations(dir string) ([]string, error) {
+func crawlTemplateLocations(dir string) ([]string, error) {
 	var out []string
 	err := filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
 		if err != nil {
 			return err
 		}
-		if d.IsDir() {
-			if _, err := os.Stat(filepath.Join(path, "spec.yaml")); err == nil {
-				out = append(out, path)
-			}
+		if !d.IsDir() {
+			return nil
+		}
+		if strings.HasSuffix(path, goldenTestDir) && d.IsDir() {
```

Comment:
The call to `d.IsDir()` isn't necessary because the `if` three lines above already guarantees that this is a directory.